src/main/scala/Exercise8_4.scala

```diff
@@ -0,0 +1,36 @@
+object Exercise8_4 extends App {
+  abstract class Item {
+    def price:Double
+    def description:String
+  }
+  class SimpleItem(val price:Double, val description:String) extends Item
+
+  class Bundle(val bundlePrice:Double, val bundleName:String) extends Item {  // An item able to store simple items
+
+    private var content:Array[SimpleItem] = new Array(0) // Stored items. I prefer not to stack bundles
+
+    def price:Double = bundlePrice + content.map(_.price).sum  // price of the bundle itself + content's total price
+
+    def description:String = { // description of bundle and its content
+      if (content.length == 0)
+        "An empty " + bundleName
+      else
+        bundleName + " with one " + content.map(_.description).mkString(", one ") + " inside."
+    }
+
+    def AddSimpleItem(newConts:SimpleItem*) { // a method to add items to the bundle
```

Comment:
does this work? I think you would need an equal
And it's one lines, I think you can delete `{}`
```suggestion
    def AddSimpleItem(newConts:SimpleItem*) = { // a method to add items to the bundle
```

Reply:
Both of them work. Why is preferable the second one?

Reply:
I would say the latter, yep.